samples/csharp_dotnetcore/51.cafe-bot/CafeBot.cs

```diff
@@ -0,0 +1,219 @@
+                    if (onTurnProperties == null)
+                    {
+                        break;
+                    }
+
+                    // Set the state with gathered properties (intent/ entities) through the onTurnAccessor
+                    await _onTurnAccessor.SetAsync(context, onTurnProperties);
+
+                    // Create dialog context.
+                    var dc = await _dialogs.CreateContextAsync(context);
+
+                    // Continue outstanding dialogs.
+                    await dc.ContinueDialogAsync();
+
+                    // Begin main dialog if no outstanding dialogs/ no one responded
+                    if (!dc.Context.Responded)
+                    {
+                        await dc.BeginDialogAsync(nameof(MainDispatcher));
+                    }
+
+                    break;
+                case ActivityTypes.ConversationUpdate:
+                    // Welcome user.
+                    await WelcomeUserAsync(context);
+                    break;
+                default:
+                    // Handle other activity types as needed.
+                    break;
+            }
+            await ConversationState.SaveChangesAsync(context);
+            await UserState.SaveChangesAsync(context);
+        }
+
+        private async Task<OnTurnProperty> DetectIntentAndEntitiesAsync(ITurnContext turnContext)
+        {
+            // Handle card input (if any), update state and return.
+            if (turnContext.Activity.Value != null)
+            {
+                var response = JsonConvert.DeserializeObject<Dictionary<string, string>>(turnContext.Activity.Value as string);
+                return OnTurnProperty.FromCardInput(response);
+            }
+
+            // Acknowledge attachments from user.
+            if (turnContext.Activity.Attachments != null && turnContext.Activity.Attachments.Count > 0)
+            {
+                await turnContext.SendActivityAsync("Thanks for sending me that attachment. I'm still learning to process attachments.");
+                return null;
+            }
+
+            // Nothing to do for this turn if there is no text specified.
+            if (string.IsNullOrWhiteSpace(turnContext.Activity.Text) || string.IsNullOrWhiteSpace(turnContext.Activity.Text.Trim()))
+            {
+                return null;
+            }
+
+            // Call to LUIS recognizer to get intent + entities
+            var luisResults = await _services.LuisServices[LuisConfiguration].RecognizeAsync(turnContext, default(CancellationToken));
+
+            // Return new instance of on turn property from LUIS results.
+            // Leverages static fromLUISResults method
+            return OnTurnProperty.FromLuisResults(luisResults);
+        }
+
+        private async Task WelcomeUserAsync(ITurnContext turnContext)
+        {
+            // Do we have any new members added to the conversation?
```

Comment:
>// Do we have any new members added to the conversation? [](start = 12, length = 56)

consider revising to "check to see if any new members were added to the conversation"